CHANGELOG.md

```diff
@@ -4,6 +4,494 @@ All notable changes to Router will be documented in this file.
+-use apollo_router::services::{RouterRequest, RouterResponse};
++use apollo_router::services::router;
+ 
+-async fn example(service: BoxService<RouterRequest, RouterResponse, BoxError>) -> RouterResponse {
++async fn example(service: router::BoxService) -> router::Response {
+-    let request = RouterRequest::builder()/*…*/.build();
++    let request = router::Request::builder()/*…*/.build();
+     service.oneshot(request).await
+ }
+```
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Some items were removed from the public API ([PR #1487](https://github.com/apollographql/router/pull/1487) [PR #1535](https://github.com/apollographql/router/pull/1535))
+
+If you used some of them and don’t find a replacement,
+please [file an issue](https://github.com/apollographql/router/issues/)
+with details about the use case.
+
+```
+apollo_router::Configuration::boxed
+apollo_router::Configuration::is_compatible
+apollo_router::errors::CacheResolverError
+apollo_router::errors::JsonExtError
+apollo_router::errors::ParsesError::print
+apollo_router::errors::PlanError
+apollo_router::errors::PlannerError
+apollo_router::errors::PlannerErrors
+apollo_router::errors::QueryPlannerError
+apollo_router::errors::ServiceBuildError
+apollo_router::json_ext
+apollo_router::layers::ServiceBuilderExt::cache
+apollo_router::mock_service!
+apollo_router::plugins
+apollo_router::plugin::plugins
+apollo_router::plugin::PluginFactory
+apollo_router::plugin::DynPlugin
+apollo_router::plugin::Handler
+apollo_router::plugin::test::IntoSchema
+apollo_router::plugin::test::MockSubgraphFactory
+apollo_router::plugin::test::PluginTestHarness
+apollo_router::query_planner::QueryPlan::execute
+apollo_router::services
+apollo_router::Schema
+```
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Router startup API changes ([PR #1487](https://github.com/apollographql/router/pull/1487))
+
+The `RouterHttpServer::serve` method and its return type `RouterHandle` were removed,
+their functionality merged into `RouterHttpServer` (formerly `ApolloRouter`).
+The builder for `RouterHttpServer` now ends with a `start` method instead of `build`.
+This method immediatly starts the server in a new Tokio task.
+
+```diff
+ RouterHttpServer::builder()
+     .configuration(configuration)
+     .schema(schema)
+-    .build()
+-    .serve()
++    .start()
+     .await
+```
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### `router_builder_fn` replaced by `shutdown` in the `Executable` builder ([PR #1487](https://github.com/apollographql/router/pull/1487))
+
+The builder for `apollo_router::Executable` had a `router_builder_fn` method
+allowing to specify how a `RouterHttpServer` (previously `ApolloRouter`) was to be created
+with a provided configuration and schema.
+The only possible variation there was specifying when the server should shut down
+with a `ShutdownSource` parameter,
+so `router_builder_fn` was replaced with a new `shutdown` method that takes that.
```

Comment:
```suggestion
The builder for `apollo_router::Executable` had a `router_builder_fn` method
allowing the specification of how a `RouterHttpServer` (previously `ApolloRouter`) was to be created
with a provided configuration and schema.
Since the only possible variation was specifying _when_ the server should shut down
(with a `ShutdownSource` parameter) the `router_builder_fn` was replaced with a new `shutdown` method.
```